client/lxd_storage_volumes.go

```diff
@@ -762,23 +762,29 @@ func (r *ProtocolLXD) MoveStoragePoolVolume(pool string, source InstanceServer,
 }
 
 // UpdateStoragePoolVolume updates the volume to match the provided StoragePoolVolume struct.
-func (r *ProtocolLXD) UpdateStoragePoolVolume(pool string, volType string, name string, volume api.StorageVolumePut, ETag string) error {
+func (r *ProtocolLXD) UpdateStoragePoolVolume(pool string, volType string, name string, volume api.StorageVolumePut, ETag string) (Operation, error) {
 	if !r.HasExtension("storage") {
-		return fmt.Errorf("The server is missing the required \"storage\" API extension")
+		return nil, fmt.Errorf("The server is missing the required \"storage\" API extension")
 	}
 
 	if volume.Restore != "" && !r.HasExtension("storage_api_volume_snapshots") {
-		return fmt.Errorf("The server is missing the required \"storage_api_volume_snapshots\" API extension")
+		return nil, fmt.Errorf("The server is missing the required \"storage_api_volume_snapshots\" API extension")
 	}
 
 	// Send the request
 	path := fmt.Sprintf("/storage-pools/%s/volumes/%s/%s", url.PathEscape(pool), url.PathEscape(volType), url.PathEscape(name))
-	_, _, err := r.query("PUT", path, volume, ETag)
+	op, _, err := r.queryOperation("PUT", path, volume, ETag, true)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	return nil
+	if r.CheckExtension("storage_volume_async_update") != nil {
+		// Wait for the operation and ignore the error. This will fail, but prevent subsequent
+		// calls from failing. Prior this extension, this API call was synchronous.
+		_ = op.Wait()
```

Comment:
@MusicDin please can you explain about why op.Wait is required here and what "prevent subsequent calls from failing" means?

Reply:
Are you trying to fake a completed operation in cases where the server is using the synchronous call?

Reply:
Yes I faked the completed operation. I missed the fact that we support RawOperations. Will update the PR.